File: CloudFlow/FrontEnd-App/src/app/shared/model/calendar/interval-calendar-event.model.ts
import { JsonObject, JsonProperty } from "json2typescript";
import { JsonDateConverter } from "../../json-mapper";
import { CalendarEventModel } from "./calendar-event.model";

@JsonObject
export abstract class IntervalCalendarEvent extends CalendarEventModel {
	
	@JsonProperty("startDate", JsonDateConverter)
	public startDate: Date = null;
	
	@JsonProperty("endDate", JsonDateConverter)
	public endDate: Date = null;
}

File: CloudFlow/src/ro/cloudSoft/cloudDoc/presentation/client/shared/model/calendar/IntervalCalendarEventModel.java
package ro.cloudSoft.cloudDoc.presentation.client.shared.model.calendar;

import java.util.Date;

public class IntervalCalendarEventModel extends CalendarEventModel {
	
	public IntervalCalendarEventModel(CalendarEventType type) {
		super(type);
	}
	
	private Date startDate;
	private Date endDate;

	public Date getStartDate() {
		return startDate;
	}
	public void setStartDate(Date startDate) {
		this.startDate = startDate;
	}
	public Date getEndDate() {
		return endDate;
	}
	public void setEndDate(Date endDate) {
		this.endDate = endDate;
	}
}
